stac_fastapi/api/stac_fastapi/api/app.py

```diff
@@ -32,6 +39,69 @@
 from stac_fastapi.types.search import STACSearch
 
 
+class VndOaiResponse(ORJSONResponse):
+    """JSON with custom, vendor content-type."""
+
+    media_type = "application/vnd.oai.openapi+json;version=3.0"
+
+
+class VndFastAPI(FastAPI):
+    """Custom FastAPI to support /api response with vendor content-type."""
+
+    def setup(self) -> None:
```

Comment:
Perhaps unnecessary code duplication can be avoided by calling `super` and then removing/re-adding the route itself.

`self` is a `Starlette` object, which has a [routes](https://github.com/encode/starlette/blob/master/starlette/applications.py#L95) property that directly returns the list of the router's routes. AFAIK adding a route appends it to the list and does no other special registration. While there's no "remove_route" method, if you find the route with the `self.openapi_url` and replace it with the one you construct with the proper response, that might save some code dupe. The worry with the dupe is that if a later version of FastAPI changes say the docs_url add logic, then this override logic won't pick up on those changes unrelated to the VndOaiResponse.

Another thing to note is that [Route](https://github.com/encode/starlette/blob/6af5c515e0a896cbf3f86ee043b88f6c24200bcf/starlette/routing.py#L189) exposes the endpoint Callable directly, so you could even wrap the endpoint in one that translates the return value into a VndOaiResponse.

Maybe not worth spending too much time trying to hack the internals of FastAPI and Starlette (though everything I mentioned is public API), but I think it would be preferable to reduce the code copy in the setup override to only as much as is needed for this small change.

Reply:
It appears as though `attrs` has a `converter` argument, which allows you to make modifications to supplied arguments. I've implemented a function which can be used as a `converter` to alter the routes such that the appropriate vendor content-type is advertised without any inheritance shenanigans